Microsoft.Toolkit.Uwp.UI.Controls/RangeSelector/RangeSelector.cs

```diff
@@ -639,19 +665,192 @@ public double StepFrequency
+        }
+
+        /// <summary>
+        /// Gets or sets the value for tick placement position.
+        /// </summary>
+        /// <value>
+        /// The value for tick placement.
+        /// </value>
+        public TickPlacement TickPlacement
+        {
+            get
+            {
+                return (TickPlacement)GetValue(TickPlacementProperty);
+            }
+
+            set
+            {
+                SetValue(TickPlacementProperty, value);
+            }
+        }
+
+        private static void TickPlacementChangedCallback(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var rangeSelector = d as RangeSelector;
+            if (rangeSelector == null)
+            {
+                return;
+            }
+
+            rangeSelector.DrawTicks();
+        }
+
+        /// <summary>
+        /// Gets or sets the value for tick offset.
+        /// </summary>
+        /// <value>
+        /// The value for tick offset.
+        /// </value>
+        public double TickOffset
+        {
+            get
+            {
+                return (double)GetValue(TickOffsetProperty);
+            }
+
+            set
+            {
+                SetValue(TickOffsetProperty, value);
+            }
+        }
+
+        private static void TickOffsetChangedCallback(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var rangeSelector = d as RangeSelector;
+            if (rangeSelector == null)
+            {
+                return;
+            }
+
+            rangeSelector.DrawTicks();
+        }
+
+        /// <summary>
+        /// Gets or sets the value for snapping to steps or tick frequency.
+        /// </summary>
+        /// <value>
+        /// The value for snapping to ticks or steps.
+        /// </value>
+        public SliderSnapsTo SnapsTo
+        {
+            get
+            {
+                return (SliderSnapsTo)GetValue(SnapsToProperty);
+            }
+
+            set
+            {
+                SetValue(SnapsToProperty, value);
+            }
+        }
+
+        private void DrawTicks()
+        {
+            if (_tickCanvas == null)
+            {
+                return;
+            }
+
+            _tickCanvas.Children.Clear();
+
+            if (TickPlacement == TickPlacement.None)
+            {
+                return;
+            }
+
+            double tickMargin = 3;
+            double tickHeight = 4;
+            double tickActualStep = _containerCanvas.ActualWidth / (Maximum - Minimum);
+            double relativeTopForInlineTicks = (_containerCanvas.ActualHeight / 2) - (tickHeight / 2);
+            double relativeTopForTopTicks = (_containerCanvas.ActualHeight / 2) - (tickMargin + tickHeight);
+            double relativeTopForBottomTicks = (_containerCanvas.ActualHeight / 2) + tickMargin;
+
+            double appliedTickFrequency = Math.Max(TickFrequency, 20 / tickActualStep);
```

Comment:
I believe the slider handles this a bit differently. It is still drawing the ticks at the actual TickFrequency, but before it renders a tick, it checks if it it's too close to the previous tick and if it is it skips rendering it.

Reply:
sounds much better, I will change it